src/main/java/core/basesyntax/lib/Injector.java

```diff
@@ -0,0 +1,36 @@
+package core.basesyntax.lib;
+
+import core.basesyntax.exceptions.NoDaoAnnotationExeption;
+import core.basesyntax.factory.Factory;
+import java.lang.reflect.Constructor;
+import java.lang.reflect.Field;
+import java.lang.reflect.InvocationTargetException;
+
+public class Injector {
+    public static Object getInstance(Class clazz) throws NoSuchMethodException,
+            IllegalAccessException, InvocationTargetException, InstantiationException {
+        Constructor constructor = clazz.getDeclaredConstructor();
+        Object instance = constructor.newInstance();
+        Field[] declaredFields = clazz.getDeclaredFields();
+        for (Field field: declaredFields) {
+            if (field.getAnnotation(Inject.class) != null) {
+                field.setAccessible(true);
+                if (field.getName().equals("betDao")) {
```

Comment:
it's better to use type instead of field name so no matter how you call you variable injector will work

Reply:
done